### PR TITLE
Adds superglue

### PR DESCRIPTION
**Pull request by unid15**
- Applying superglue to an item makes you unable to drop it!!!
- Superglued beakers can't be put into chem dispensers, etc. I went through almost everything don't worry :+1: :memo:
- Only way to become free is to cut your hands off or wait 15 minutes 
- Traitor clowns can buy this for 4 telecrystals

200 changed files because most of the code in which _drop_item()_ occurred has been changed to only work if that drop_item() returned 1.

### CODE REVIEW

File: code/setup.dm [hunk 1]
```diff
@@ -1337,3 +1337,6 @@ var/proccalls = 1
 #define MODE_CULTCHAT "cultchat"
 #define MODE_ANCIENT "ancientchat"
 
+//For the canremove variable on items:
+#define CANT_REMOVE_NORMAL	1 //Can't remove when equipped, can hold and drop just fine!
+#define CANT_REMOVE_GLUE	2 //Can't remove, can't drop! Glued to your hands, basically
```
Comment (Iamgoofball) [inline code/setup.dm]: so wait

you ported nodrop

but hardcoded it for glue

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -34,7 +34,10 @@
 	var/permeability_coefficient = 1 // for chemicals/diseases
 	siemens_coefficient = 1 // for electrical admittance/conductance (electrocution checks and shit) - 0 is not conductive, 1 is conductive - this is a range, not binary
 	var/slowdown = 0 // How much clothing is slowing you down. Negative values speeds you up
+
 	var/canremove = 1 //Mostly for Ninja code at this point but basically will not allow the item to be removed if set to 0. /N
+	var/glued = 0 //If 1, can't drop it from hands!
```
Comment (Iamgoofball) [inline code/game/objects/items.dm]: oh god this is horrible

**Comment by Iamgoofball** [comment]
https://github.com/tgstation/-tg-station/pull/2772

port this instead and axe 99% of this shit, this is a horrible way to go about doing NODROP and hardcoding is is bad

**Comment by Zth--** [comment]
wow WOW ALL THOSE CHANGES

**Comment by ihadtoregisterforthis** [comment]
Why not make it mime specific for now?

**Comment by Arthurdentist** [comment]
Will you be able to apply this to clothing so they can't take that off either?

**Comment by DeityLink** [comment]
I read about a similar idea earlier in the thread. I vaguely thought about coding it by making use of the already existing "canremove" item var. Can't you use that one instead of adding another var?

**Comment by Iamgoofball** [comment]
or instead port the tg implementation of NODROP which axes canremove and works much much better

**Comment by unid15** [comment]
That var only prevents unequipping items, not dropping or throwing them.
It'd be much more boring that way
On 19 Oct 2015 04:06, "DeityLink" <notifications@github.com> wrote:
>
> I read about a similar idea earlier in the thread. I vaguely thought
about coding it by making use of the already existing "canremove" item var.
Can't you use that one instead of adding another var?

**Comment by unid15** [comment]
>hardcoding
No hardcode here. All the changes are here to make shit like putting a
beaker inside a chemdispenser not work if drop_item returned 0 (which is
how it should've been In the first place).

**Comment by Iamgoofball** [comment]
You realize unequip is called by drop and throw, right?

We use it for armblades and shit at tg

**Comment by Probe1** [comment]
You edited 10% of all the files to add superglue?

**Comment by unid15** [comment]
Yeah I did

But think of it not as "edited 10% of files to add one item", but more as
"edited 10% of the files to add a new functionality with an item on top of
it", since with this, nondroppable items can exist.

**Comment by Iamgoofball** [comment]
maybe instead of **hardcoding this to glue you should listen to what I'm saying and replace your shitty hardcoded glue mess with NODROP**

**Comment by ihadtoregisterforthis** [comment]
A lot of line changes but looks like there wasn't any other way.

Maybe for a later feature, they could work as longer buckling to a chair or sticking someone to the floor.

**Comment by Iamgoofball** [comment]
the change numbers isn't avoidable but FUCK nodrop is better **JUST FUCKING PORT NODROP**

**Comment by unid15** [comment]
What's the difference between nodrop AnD the way I've done it?

**Comment by N3X15** [comment]
Lol @iamgoofball telling people how to do better quality code

**Comment by Iamgoofball** [comment]
That's why I'm not shilling my code, and shilling other people's better code

**Comment by Fox-McCloud** [comment]
We have NODROP on Para, and I can confirm that it's functionally vastly superior to can_remove---hell, in a PR that implements Bay's hardsuits, NODROP even resolved a few problems Bay couldn't resolved with can_remove. It's a very very good system.

15 minutes seems like an awfully long time though; even 5-8 minutes is pretty crazy.

**Comment by Arthurdentist** [comment]
super glue should be eternal. If you want to drop the item have your arm taken off.

Honk

**Comment by unid15** [comment]
git is fucking my shit up

**Comment by Probe1** [comment]
you're gonna re-open right?

**Comment by unid15** [comment]
yes

**Comment by Probe1** [comment]
a true coder and a real human bean.

**Pull request closed without merging**